packages/devtools_app/lib/src/shared/preferences.dart

```diff
@@ -114,25 +131,54 @@ class InspectorPreferencesController extends DisposableController
     });
   }
 
-  void _initCustomPubRootListeners() {
-    // TODO(CoderDake): add _customPubRootDirectories listener back when
-    // finalizing https://github.com/flutter/devtools/issues/3941
-    /*
-    addAutoDisposeListener(_customPubRootDirectories, () {
```

Comment:
is it no longer necessary to listen to `_customPubRootDirectories`?

Reply:
Correct, this is now only done explicitly when we call _persistCustomPubRootDirectoriesToStorage.
The upside to this is that our values in storage will be more accurate, The downside is that we may redundantly save the same value a couple of times.